### PR TITLE
mgr/snap_schedule: fix message format error

### PR DESCRIPTION
**Pull request by mchangir**
* correction to the error message
* added a new test case to test the rejection of duplicate retention

Fixes: https://tracker.ceph.com/issues/64042
Signed-off-by: Milind Changire <mchangir@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [x] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Review by rishabh-d-dave — APPROVED** [review]

**Comment by mchangir** [comment]
jenkins test api

**Review by dparmar18 — APPROVED** [review]

File: src/pybind/mgr/snap_schedule/fs/schedule.py [hunk 1]
```diff
@@ -364,9 +364,9 @@ def add_retention(cls,
             current_retention = json.loads(current)
             for r, v in retention.items():
                 if r in current_retention:
-                    msg = (f'Retention for {r} is already present with value'
-                           f'{current_retention[r]}. Please remove first')
-                    raise ValueError(msg)
+                    msg = (f'Retention for {r} is already present with value '
+                           f'{current_retention[r]}. Please remove it first.')
+                    raise FileExistsError(msg)
```
Comment (vshankar) [inline src/pybind/mgr/snap_schedule/fs/schedule.py:369]: Would this need a QA change anywhere?

Comment (vshankar) [reply src/pybind/mgr/snap_schedule/fs/schedule.py:369]: @mchangir ping?

Comment (mchangir) [reply src/pybind/mgr/snap_schedule/fs/schedule.py:369]: > Would this need a QA change anywhere?

yeah, adding a test case would be apt
I'll get that done

Comment (mchangir) [reply src/pybind/mgr/snap_schedule/fs/schedule.py:369]: test case added

Comment (dparmar18) [inline src/pybind/mgr/snap_schedule/fs/schedule.py:369]: can we catch errno 17 here? so that we're sure it has sent `FileExistsError`

Comment (dparmar18) [reply src/pybind/mgr/snap_schedule/fs/schedule.py:369]: oh sorry i meant to point at the test case not source code.

File: qa/tasks/cephfs/test_snap_schedules.py [hunk 2]
```diff
@@ -545,6 +545,30 @@ def test_snap_schedule_for_number_of_snaps_retention(self):
...
+
+        self.mount_a.run_shell(['mkdir', '-p', test_dir[1:]])
+
+        self.fs_snap_schedule_cmd('add', path=test_dir, snap_schedule='1m')
+        self.fs_snap_schedule_cmd('retention', 'add', path=test_dir,
+                                  retention_spec_or_period='m',
+                                  retention_count='50')
+
+        # Adding a duplicate retention spec should fail
+        with self.assertRaises(CommandFailedError):
```
Comment (dparmar18) [inline qa/tasks/cephfs/test_snap_schedules.py:562]: here

Comment (neesingh-rh) [reply qa/tasks/cephfs/test_snap_schedules.py:562]: Agree! We can add someting like: `self.assertEqual(exception status, code)` to do that.

Comment (vshankar) [reply qa/tasks/cephfs/test_snap_schedules.py:562]: Sounds like what @mchangir is telling here makes sense. @dparmar18 @neesingh-rh What's pending?

**Pull request closed without merging**